platform/linux-dpdk/include/odp_packet_io_internal.h

```diff
@@ -1,157 +0,0 @@
-	odp_ticketlock_t txl;		/**< TX ticketlock */
-	int cls_enabled;		/**< is classifier enabled */
-	odp_pktio_t handle;		/**< pktio handle */
-	enum {
-		/* Not allocated */
-		PKTIO_STATE_FREE = 0,
-		/* Close pending on scheduler response. Next state after this
-		 * is PKTIO_STATE_FREE. */
-		PKTIO_STATE_CLOSE_PENDING,
-		/* Open in progress.
-		   Marker for all active states following under. */
-		PKTIO_STATE_ACTIVE,
-		/* Open completed */
-		PKTIO_STATE_OPENED,
-		/* Start completed */
-		PKTIO_STATE_STARTED,
-		/* Stop pending on scheduler response */
-		PKTIO_STATE_STOP_PENDING,
-		/* Stop completed */
-		PKTIO_STATE_STOPPED
-	} state;
-	odp_pktio_config_t config;	/**< Device configuration */
-	classifier_t cls;		/**< classifier linked with this pktio*/
-	odp_pktio_stats_t stats;	/**< statistic counters for pktio */
-	char name[PKTIO_NAME_LEN];	/**< name of pktio provided to
-					   pktio_open() */
-	odp_pool_t pool;
-	odp_pktio_param_t param;
-
-	/* Storage for queue handles
-	 * Multi-queue support is pktio driver specific */
-	unsigned num_in_queue;
-	unsigned num_out_queue;
-
-	struct {
-		odp_queue_t        queue;
-		queue_t            queue_int;
-		odp_pktin_queue_t  pktin;
-	} in_queue[PKTIO_MAX_QUEUES];
-
-	struct {
-		odp_queue_t        queue;
-		odp_pktout_queue_t pktout;
-	} out_queue[PKTIO_MAX_QUEUES];
-};
-
-union pktio_entry_u {
-	struct pktio_entry s;
-	uint8_t pad[ROUNDUP_CACHE_LINE(sizeof(struct pktio_entry))];
-};
-
-typedef struct {
-	odp_spinlock_t lock;
-	pktio_entry_t entries[ODP_CONFIG_PKTIO_ENTRIES];
-} pktio_table_t;
-
-extern void *pktio_entry_ptr[];
-
-static inline int pktio_to_id(odp_pktio_t pktio)
-{
-	return _odp_typeval(pktio) - 1;
-}
-
-static inline pktio_entry_t *get_pktio_entry(odp_pktio_t pktio)
-{
-	if (odp_unlikely(pktio == ODP_PKTIO_INVALID))
-		return NULL;
-
-	if (odp_unlikely(_odp_typeval(pktio) > ODP_CONFIG_PKTIO_ENTRIES)) {
-		ODP_DBG("pktio limit %d/%d exceed\n",
-			_odp_typeval(pktio), ODP_CONFIG_PKTIO_ENTRIES);
-		return NULL;
-	}
-
-	return pktio_entry_ptr[pktio_to_id(pktio)];
-}
-
-static inline int pktio_cls_enabled(pktio_entry_t *entry)
-{
-	return entry->s.cls_enabled;
-}
-
-static inline void pktio_cls_enabled_set(pktio_entry_t *entry, int ena)
-{
-	entry->s.cls_enabled = ena;
-}
-
-/*
- * Dummy single queue implementations of multi-queue API
- */
-int single_input_queues_config(pktio_entry_t *entry,
-			       const odp_pktin_queue_param_t *param);
-int single_output_queues_config(pktio_entry_t *entry,
-				const odp_pktout_queue_param_t *param);
-int single_recv_queue(pktio_entry_t *entry, int index, odp_packet_t packets[],
-		      int num);
-int single_send_queue(pktio_entry_t *entry, int index,
-		      const odp_packet_t packets[], int num);
-
-int pktin_poll_one(int pktio_index,
-		   int rx_queue,
-		   odp_event_t evt_tbl[]);
-int pktin_poll(int pktio_index, int num_queue, int index[]);
-void pktio_stop_finalize(int pktio_index);
-
-#ifdef __cplusplus
-}
-#endif
-
-#endif
```

Comment:
It's somewhat disconcerting to see large chunks of code being removed without any obvious replacement. Commit log should explain why this is being done and where the duplicate code is located.

Reply:
Agree. I understand the change. But it will help everyone if we have explanation in the commit log.

Reply:
Verbose commit logs added in v2